Dbluhm/feature/mediator coordination

burdettadam opened the pull request:
sync up pr. checkout what I have been doing.

ianco committed:
How to run demo with efk

Signed-off-by: Ian Costanzo <ian@anon-solutions.ca>

ianco committed:
Merge branch 'master' of https://github.com/ianco/aries-cloudagent-python

ianco committed:
Merge branch 'master' of https://github.com/ianco/aries-cloudagent-python

ianco committed:
Merge branch 'master' of https://github.com/ianco/aries-cloudagent-python

ianco committed:
Merge branch 'master' of https://github.com/ianco/aries-cloudagent-python

ianco committed:
Merge remote-tracking branch 'nick/hotfix-635'

ianco committed:
Merge branch 'master' of https://github.com/ianco/aries-cloudagent-python

ianco committed:
Merge branch 'master' of https://github.com/ianco/aries-cloudagent-python

ianco committed:
A few pwd details for the mobile demo

Signed-off-by: Ian Costanzo <ian@anon-solutions.ca>

ianco committed:
A few pwd details for the mobile demo

Signed-off-by: Ian Costanzo <ian@anon-solutions.ca>

ianco committed:
A few pwd details for the mobile demo

Signed-off-by: Ian Costanzo <ian@anon-solutions.ca>

ianco committed:
Merge branch 'master' of https://github.com/ianco/aries-cloudagent-python

andrewwhitehead committed:
update version to 0.5.5

Signed-off-by: Andrew Whitehead <cywolf@gmail.com>

andrewwhitehead committed:
Merge pull request #750 from andrewwhitehead/pre-release-0.5.5

Update version to 0.5.5

ianco committed:
Merge remote-tracking branch 'upstream/master'

esune committed:
Ensure no attempt to publish endpoints is made when in read-only mode

Signed-off-by: Emiliano Suñé <emiliano.sune@gmail.com>

andrewwhitehead committed:
Merge pull request #758 from esune/master

Ensure no attempt to publish endpoints is made when in read-only mode

andrewwhitehead committed:
switch to official black github action

Signed-off-by: Andrew Whitehead <cywolf@gmail.com>

swcurran committed:
Merge pull request #759 from andrewwhitehead/fix/black-check

Switch to official black github action

andrewwhitehead committed:
update version to 0.5.6

Signed-off-by: Andrew Whitehead <cywolf@gmail.com>

burdettadam committed:
Update demo readme to suggest the correct flag to enable routing during the performance demo.
Here is a link to the argument configuration, https://github.com/hyperledger/aries-cloudagent-python/blob/master/demo/runners/performance.py#L544

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

andrewwhitehead committed:
Merge pull request #760 from andrewwhitehead/pre-release-0.5.6

Update version to 0.5.6

ianco committed:
Merge remote-tracking branch 'upstream/master'

ianco committed:
Merge branch 'master' of https://github.com/ianco/aries-cloudagent-python

sklump committed:
work in progress

Signed-off-by: sklump <srklump@hotmail.com>

sklump committed:
convert short verkeys from ledger to full before use

Signed-off-by: sklump <srklump@hotmail.com>

andrewwhitehead committed:
Merge branch 'master' into update-demo/readme

swcurran committed:
Merge pull request #761 from burdettadam/update-demo/readme

Update demo readme …

andrewwhitehead committed:
Merge branch 'master' into full-verkey-on-public-did-invi

andrewwhitehead committed:
Merge pull request #762 from sklump/full-verkey-on-public-did-invi

Full verkey on public did invi

burdettadam committed:
broken paths for admin api. oct22

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

sklump committed:
commands/start unit tests

Signed-off-by: sklump <srklump@hotmail.com>

burdettadam committed:
rest api before refactor to http endpoints.

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
mediation coordination admin http endpoints, untested, get single, list, create

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

sklump committed:
nudge floats to ints in non-revocation intervals

Signed-off-by: sklump <srklump@hotmail.com>

sklump committed:
black tweak

Signed-off-by: sklump <srklump@hotmail.com>

andrewwhitehead committed:
Merge branch 'master' into master

andrewwhitehead committed:
Merge pull request #749 from ianco/master

Updates to the mobile demo for PWD

andrewwhitehead committed:
Merge branch 'master' into unit-tests-commands

andrewwhitehead committed:
Merge pull request #763 from sklump/unit-tests-commands

commands/start unit tests

andrewwhitehead committed:
Merge branch 'master' into nudge-floats-to-ints-in-non-revocation-intervals

andrewwhitehead committed:
Merge pull request #764 from sklump/nudge-floats-to-ints-in-non-revocation-intervals

Nudge floats to ints in non revocation intervals

sklump committed:
allow interval in holder cred rev check via ledger

Signed-off-by: sklump <srklump@hotmail.com>

sklump committed:
Merge branch 'master' of https://github.com/hyperledger/aries-cloudagent-python into allow-interval-in-holder-check-cred-rev-status

burdettadam committed:
before testing.

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

dbluhm commented on aries_cloudagent/protocols/coordinate_mediation/v1_0/manager.py:
```diff
@@ -75,16 +75,27 @@ async def _create_routing_did(self) -> DIDInfo:
         await record.save(self.context, reason="New mediation request received")
         return record
-
+    
```
Trailing whitespace here... mind cleaning that up?

burdettadam replied:
I will clean that up right away.

dbluhm commented on aries_cloudagent/protocols/coordinate_mediation/v1_0/models/mediation_record.py:
```diff
@@ -29,12 +29,16 @@ class Meta:
     STATE_DENIED = "denied"
-
+    
+    ORIGINATION_INTERNAL = "internal"
+    ORIGINATION_EXTERNAL = "external"
```
I wonder if it might be clearer to express this as "role" rather than "origination" -- so client-server vs internal-external. Exactly what internal/external means isn't immediately obvious.

burdettadam replied:
I agree I will make the change.

dbluhm commented on aries_cloudagent/protocols/routing/v1_0/routes.py:
```diff
@@ -0,0 +1,38 @@
+""" routing routes for did_comm forwarding"""
+
+from aiohttp import web
+from aiohttp_apispec import (
+    docs,
+    match_info_schema,
+    querystring_schema,
+    request_schema,
+    response_schema,
+)
+
+from .message_types import SPEC_URI
+
+async def register(app: web.Application):
+    """Register routes."""
+
+    app.add_routes(
+        [
+            #web.get("/routess", list_keylists, allow_head=False),
+            #web.update("/routes", update_keylists ),
+            #web.delete("/routes", delete_keylist ),
+        ]
+    )
+
+
+def post_process_routes(app: web.Application):
+    """Amend swagger API."""
+
+    # Add top-level tags description
+    if "tags" not in app._state["swagger_dict"]:
+        app._state["swagger_dict"]["tags"] = []
+    app._state["swagger_dict"]["tags"].append(
+        {
+            "name": "routes",
+            "description": "List of connection ID to key list mappings used for message forwarding",
+            "externalDocs": {"description": "Specification", "url": SPEC_URI},
+        }
+    )
```
This isn't defining any routes, is it? Is this a placeholder for intended work?

burdettadam replied:
yes, place holder, no actual routes defined yet.

dbluhm reviewed (changes requested):
A few small comments. Also, do you have flake8 activated in your environment/editor? ACA-Py's CI/CD pipeline will fail on flake8 "errors" so we should make sure we're addressing those as we go. I think it's picky about inline comments, for instance, requiring two spaces between the code and the `#` of the comment.

sklump committed:
set auto-remove consistently in cred ex records

Signed-off-by: sklump <srklump@hotmail.com>

bcersows committed:
Add support for custom event web hooks

Enhance `agent.py` to read env variable `WEBHOOK_TARGET`, which will be forwarded as a `--webhook-url` to aca-py.
`run_demo` received a new option `--webhook-url`, which (together with env variable `WEBHOOK_URL`) will forward the URL to the agent.

Example usage: `WEBHOOK_URL=host.docker.internal:8888 ./run_demo faber --webhook-url`

Signed-off-by: Björn Cersowsky <bce@crossconsense.de>

bcersows committed:
Merge branch 'master' of https://github.com/bcersows/aries-cloudagent-python

burdettadam committed:
happier flake8, role added to record

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
start of granted handler

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

dbluhm commented:
These changes are looking good! Thanks! I noticed you started on the grant handler -- should I wait to merge this PR?

sklump committed:
address issues 769, 771

Signed-off-by: sklump <srklump@hotmail.com>

andrewwhitehead committed:
Merge pull request #766 from sklump/cred-ex-record-deletion-audit

set auto-remove consistently in cred ex records

andrewwhitehead committed:
Merge branch 'master' into non-revocation-interval-detail

andrewwhitehead committed:
Merge pull request #772 from sklump/non-revocation-interval-detail

address issues 769, 771

andrewwhitehead committed:
Merge branch 'master' into master

sklump committed:
sync base with indy in holder/

Signed-off-by: sklump <srklump@hotmail.com>

sklump committed:
merge pydoc

Signed-off-by: sklump <srklump@hotmail.com>

andrewwhitehead committed:
Merge pull request #767 from bcersows/master

Enable to set custom web hook in demo

burdettadam committed:
cleaner code.

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

andrewwhitehead committed:
Merge branch 'master' into allow-interval-in-holder-check-cred-rev-status

andrewwhitehead committed:
Merge pull request #765 from sklump/allow-interval-in-holder-check-cred-rev-status

allow interval in holder cred rev check via ledger

burdettadam commented:
using the swagger admin API I had tested, by hand, up to receiving a granted message. At which point things broke because aca-py did not have a handler, so I added one. 
I still need to add to the admin API endpoints for keylist management. You can merge now or wait, I just want to make sure I am coding in plain sight and we don't end up coding the same stuff twice.

burdettadam committed:
less buggy admin api for meatiators

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

dbluhm commented on aries_cloudagent/protocols/coordinate_mediation/v1_0/models/mediation_record.py:
```diff
@@ -84,3 +90,77 @@ class Meta:
+    ),
+    required=False,
+    description="List of mediator rules for recipient",
+)
+
+
+RECIPIENT_TERMS_SCHEMA = fields.List(
+    fields.Str(
+        description="Indicate terms that the recipient "
+        "requires the mediator to agree to"
+    ),
+    required=False,
+    description="List of recipient rules for mediation",
+)
+
+
+ENDPOINT_SCHEMA = fields.Str(
+    description="endpoint on which messages destined "
+    "for the recipient are received.",
+    example="http://192.168.56.102:8020/"
+)
+
+
+ROUTING_KEYS_SCHEMA = fields.List(
+    fields.Str(
+        description="Keys to use for forward message packaging"
+    ),
+    required=False,
+)
+
+
+class MediationRecordReportSchema(OpenAPISchema):
+    """MediationRecordSchema schema."""
+
+    mediation_id = MEDIATION_ID_SCHEMA
+    conn_id = CONNECTION_ID_SCHEMA
+    mediator_terms = MEDIATOR_TERMS_SCHEMA
+    recipient_terms = RECIPIENT_TERMS_SCHEMA
+    endpoint = ENDPOINT_SCHEMA
+    routing_keys = ROUTING_KEYS_SCHEMA
```
I'm confused what this is intended for. Care to clarify?

burdettadam replied:
I found the logic used to validate the admin swagger API parameters to be used in multiple places, so I refactored the schema logic into the model. I did not see a better place for it. I could move it back into routes.py if needed.

dbluhm replied:
I think splitting it out is good but lets put it in a new file in models.

burdettadam replied:
done!

burdettadam committed:
start of adding update keylist params to admin api.

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

dbluhm commented on aries_cloudagent/protocols/coordinate_mediation/v1_0/manager.py:
```diff
@@ -151,6 +189,11 @@ async def get_keylist(self, record: MediationRecord) -> Sequence[RouteRecord]:
         route_mgr = RoutingManager(self.context)
         return await route_mgr.get_routes(record.connection_id)
 
+    async def create_keylist(self, record: MediationRecord, did: DIDInfo) -> RouteRecord:
+        """Retrieve routes for connection."""
+        route_mgr = RoutingManager(self.context)
+        return await route_mgr.create_route_record(record.connection_id, did.verkey)
+
```
I'm suspicious of this method. Might just be the incorrect doc string.

burdettadam replied:
"kinda sus!" I will update the document string, but this method may not be needed anymore. should we remove it?

dbluhm commented on aries_cloudagent/protocols/coordinate_mediation/v1_0/manager.py:
```diff
@@ -99,6 +126,17 @@ async def grant_request(self, mediation: MediationRecord) -> MediationGrant:
         )
         return grant
 
+    async def granted_request(self, mediation: MediationRecord,
+                              endpoint,
+                              routing_did_verkey: Sequence[str]
+                              ) -> MediationRecord:
+        """update Mediation state to granted."""
+        # TODO: ?create if not existing?
+        mediation.state = MediationRecord.STATE_GRANTED
+        await mediation.save(self.context, reason="Mediation request granted")
+        # TODO: update endpoint/keylists from parameters
+        return mediation
+
```
For now, let's leave this as a stub.

burdettadam replied:
I need this for mediation_granted_handler.py where a client aca-py has received a grant message and needs to update his state. 
Is there something you don't like I can address with the method? I could move the logic directly into the handler but that will break the manager pattern we have.

dbluhm commented on aries_cloudagent/protocols/coordinate_mediation/v1_0/manager.py:
```diff
@@ -74,11 +75,37 @@ async def _create_routing_did(self) -> DIDInfo:
-    async def receive_request(self, request: MediationRequest) -> MediationRecord:
-        """Create a new mediation record to track this request."""
+    async def receive_request(self,
+                              conn_id: str,
+                              request: MediationRequest
+                              ) -> MediationRecord:
+        """Create a new mediation record to track external request."""
+        role = MediationRecord.ROLE_SERVER
+        record = await self._store_request(request, conn_id, role)
+        return record
+
+    async def prepare_request(self,
+                              conn_id: str,
+                              request: MediationRequest
+                              ) -> MediationRecord:
+        """Create a new mediation record to track internal request."""
+        role = MediationRecord.ROLE_CLIENT
+        record = await self._store_request(request, conn_id, role)
+        return record
+
+    async def _store_request(self,
+                             request: MediationRequest,
+                             _conn_id: str,
+                             _role: str
+                             ) -> MediationRecord:
+        """Create a new mediation record to track a request."""
         # TODO: Determine if terms are acceptable
+        connection_record = await ConnectionRecord.retrieve_by_id(
+            self.context, _conn_id
+        )
         record = MediationRecord(
-            connection_id=self.context.connection_record.connection_id,
+            role=_role,
+            connection_id=connection_record.connection_id,
```
I think I prefer a little redundancy to having this `_store_request` method being split out.

burdettadam replied:
I split the logic out because it would be a direct copy-paste of the same code inside recieve_request and prepare_request. We could have redundancy or even push the logic into the routes.py handler. which do you prefer?

dbluhm replied:
The point I take issue with is that you've condensed just two method calls into one method call. It is several lines but in this case again, I prefer the redundancy and clarity.

burdettadam replied:
I understand. I will change this back.

dbluhm commented on aries_cloudagent/protocols/coordinate_mediation/v1_0/manager.py:
```diff
@@ -74,11 +75,37 @@ async def _create_routing_did(self) -> DIDInfo:
+        record = await self._store_request(request, conn_id, role)
+        return record
+
+    async def prepare_request(self,
+                              conn_id: str,
+                              request: MediationRequest
+                              ) -> MediationRecord:
+        """Create a new mediation record to track internal request."""
+        role = MediationRecord.ROLE_CLIENT
+        record = await self._store_request(request, conn_id, role)
+        return record
```
Let's leave this stubbed out for now.

burdettadam replied:
I need this for routes.py. Should I move the logic into the admin handler?

dbluhm replied:
As per our offline discussion, I have some internal conflict on this point but for now I'll concede. Carry on!

dbluhm commented on aries_cloudagent/protocols/coordinate_mediation/v1_0/handlers/mediation_granted_handler.py:
```diff
@@ -0,0 +1,43 @@
+                endpoint=endpoint,
+                routing_did_verkey=routing_keys
+            )
+        except StorageNotFoundError as err:
+            pass
+        else:
+            pass  # if not existing record, do nothing
+            # TODO: ?create if not existing?
```
We should probably send a `problem report` message to the sender saying or just raise a handler exception like on line 26.

burdettadam replied:
I agree. I will add the error notification.

dbluhm commented on aries_cloudagent/protocols/coordinate_mediation/v1_0/handlers/mediation_denied_handler.py:
```diff
@@ -0,0 +1,42 @@
+            _record = await MediationRecord.retrieve_by_id(
+                context, context.mediation_id
+            )
+            await mgr.deny_request(
+                mediation=_record,
+                mediator_terms=context.mediator_terms,
+                recipient_terms=context.recipient_terms
+            )
```
`MediationManager.deny_request` is a server role method, not client. We should make the method names more explicit if necessary to clearly denote that. This usage of it here is incorrect.

burdettadam replied:
How is this incorrect use? deny_request updates state and create a denied message, while mediation_denied_handler.py does not use the created denied message, it does behave as desired. Should we separate the state update logic to be in the denied request handler? Maybe create new methods in the manager for updating the state?

dbluhm replied:
I think the real problem is the semantics of it. That sort of conceptual overlap will just be misleading and confusing, especially given that we already know that keeping straight what operations pertain to what role is difficult. This is a case where redundancy is preferable because of the further provided clarity.

Additionally, switching the state directly would take just as many lines as calling this method does anyways, fewer even! :P
```python
_record.state = MediationRecord.STATE_DENIED
await _record.save(context, reason='Client mediation request was denied')
```

burdettadam replied:
Astute observation, I will update the denied handler with the needed record update.

dbluhm commented on aries_cloudagent/protocols/coordinate_mediation/v1_0/handlers/mediation_denied_handler.py:
```diff
@@ -0,0 +1,42 @@
+                mediator_terms=context.mediator_terms,
+                recipient_terms=context.recipient_terms
+            )
+        except StorageNotFoundError as err:
+            pass
+        else:
+            pass  # if not existing record, do nothing
+            # TODO: ?create if not existing?
```
Send problem report or raise handler exception as on line 27.

burdettadam replied:
I think the question can be phrased as, "what should aca-py do when it receives a mediation denied message for a connection with no mediation?". Currently, aca-py quietly does nothing, we could send a problem report back on the connection, but rfc-0211 does not define problem reports, raising an error does not feel correct because nothing has errored. What is the desired behavior?

dbluhm replied:
We're already raising an exception on line 27 when the connection isn't in an acceptable state so I'm not sure what the difference is here. The RFC probably should have problem reports defined but problem reports are a "courtesy" and are not strictly speaking required. It's up to the agent developer and the use case to determine how chatty agents are. If you're really against raising the error (which just gets spit out by the dispatcher to the logs anyways), then I would recommend logging something at least.

burdettadam replied:
I see your point and agree. I will add the needed report and exception.

burdettadam committed:
small document string change

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
terrible bug, beware

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
all admin handlers need to be in routes to work

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

olichwiruk committed:
Fix typo in webhook example path

Signed-off-by: Marcin <marcin@opensoftware.pl>

burdettadam committed:
removed un-needed import

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

ianco committed:
Merge pull request #775 from THCLab/fix_typo

Fix typo in webhook example path

burdettadam committed:
passing tests

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
passing tests

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
black code formatter check, fix.

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
black code formatter check, fix.

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
Merge branch 'master' into invitation-connection-target

burdettadam committed:
Merge branch 'invitation-connection-target' of github.com:burdettadam/aries-cloudagent-python into invitation-connection-target

burdettadam committed:
merging upstream

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
start of mediated invitation

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
Merge branch 'feature/mediator-coordination' of github.com:dbluhm/aries-cloudagent-python into dbluhm/feature/mediator-coordination

burdettadam committed:
fixed lost merge from master

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
more stable getters

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
Merge branch 'feature/mediator-coordination' of github.com:dbluhm/aries-cloudagent-python into dbluhm/feature/mediator-coordination

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
reconciliation, code corrections from pr feedback and merged branches.

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
update grant state bug

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
fixed update grant state

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

sklump committed:
query strings are always strings

Signed-off-by: sklump <srklump@hotmail.com>

burdettadam committed:
before pep8 fix

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
happier flake8

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

sklump committed:
strip explicit initiator from, plumb rfc23 preparation into connection records

Signed-off-by: sklump <srklump@hotmail.com>

sklump committed:
perfunctory change so CI can re-run

Signed-off-by: sklump <srklump@hotmail.com>

andrewwhitehead committed:
Merge pull request #782 from sklump/cred-count-int-fix

query strings are always strings

andrewwhitehead committed:
Merge branch 'master' into invitation-connection-target

andrewwhitehead committed:
Merge pull request #778 from burdettadam/invitation-connection-target

feature/Connection Invitation with routingKeys

andrewwhitehead committed:
remove update_record_tags, delete_record_tags; add find_record as a better alternative to search_records(..) -> fetch_one

Signed-off-by: Andrew Whitehead <cywolf@gmail.com>

sklump committed:
it is possible to post null body

Signed-off-by: sklump <srklump@hotmail.com>

sklump committed:
work in progress

Signed-off-by: sklump <srklump@hotmail.com>

sklump committed:
Merge branch 'master' of https://github.com/hyperledger/aries-cloudagent-python into adapt-conn-recs-for-rfc23

sklump committed:
Merge pull request #784 from andrewwhitehead/storage-api-1

Initial storage API adjustments

sklump committed:
base record post-filter by alternatives, positive and negative

Signed-off-by: sklump <srklump@hotmail.com>

sklump committed:
Merge branch 'master' of https://github.com/hyperledger/aries-cloudagent-python into adapt-conn-recs-for-rfc23

burdettadam committed:
Merge branch 'master' of github.com:hyperledger/aries-cloudagent-python into dbluhm/feature/mediator-coordination

andrewwhitehead committed:
Merge branch 'master' into invitation-body-optional

andrewwhitehead committed:
Merge pull request #786 from sklump/invitation-body-optional

it is possible to post null body

sklump committed:
tweak connection records to work for RFC 23

Signed-off-by: sklump <srklump@hotmail.com>

sklump committed:
hush lgtm

Signed-off-by: sklump <srklump@hotmail.com>

andrewwhitehead committed:
Merge branch 'master' into adapt-conn-recs-for-rfc23

andrewwhitehead committed:
Merge pull request #787 from sklump/adapt-conn-recs-for-rfc23

Adapt conn recs for rfc23

burdettadam committed:
Merge branch 'master' of github.com:hyperledger/aries-cloudagent-python into dbluhm/feature/mediator-coordination

burdettadam committed:
less bugs

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
automagical

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

sklump committed:
store conn records with legacy values for their-role, state

Signed-off-by: sklump <srklump@hotmail.com>

ianco committed:
Merge pull request #790 from sklump/write-conn-rec-enums-as-old-values

store conn records with legacy values for their-role, state

burdettadam committed:
update broker side of things.

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

burdettadam committed:
Merge branch 'master' of github.com:hyperledger/aries-cloudagent-python into dbluhm/feature/mediator-coordination

burdettadam committed:
set correct argument vars

Signed-off-by: Adam Burdett <burdettadam@gmail.com>
